### ISSUE TITLE
Applying Issue

### ISSUE THREAD

**Issue by zy937085318** [MarvinTeichmann/ConvCRF#7]
Hi, I'm quite interested in CRF like models in segmentation and contour detection tasks. I think the ConvCRF is suitable for contour detection much, so I link it to the end of my naive CNN model, but when I set the hyperparameters as listed in the paper, the final prediction of ConvCRF would not change in 200 epochs. While I increase the learning rate to 1e-3, the prediction would generally be the same as the unary.  By the way, I just apply the default_config and change the trainable mark to True. Is there any other detail I need to add? Thx for ur great work:)

**Comment by MarvinTeichmann** [comment]
Hi, thanks for your interest in my work. Setting trainable to true and then backprop the gradients through the CRF is all which should be needed to train the CRF. 

You mentioned that the prediction is always the same as the unary. I believe that this might be the actual issue. If the CRF is not doing anything, then the gradients are zero and learning will not doe anything. Make sure that during early iterations the CRF is augmenting the prediction at least slightly (it does not need to change the argmax, but it has to modify the floating point values significantly).

My guess would be that this is an initialization error. I would recommend playing around with the variances [sdims, schan, compat]. Also [have a look](https://github.com/MarvinTeichmann/ConvCRF/blob/master/benchmark.py#L63) at my discussion regarding recommended variances for normalized images. In general, make sure that at [this line](https://github.com/MarvinTeichmann/ConvCRF/blob/master/convcrf/convcrf.py#L599) the message is not negligible small compared to the unary.
